nowcasting_dataset/data_sources/metadata/metadata_model.py

```diff
@@ -1,11 +1,96 @@
 """ Model for output of general/metadata data, useful for a batch """
-from nowcasting_dataset.data_sources.datasource_output import DataSourceOutput
 
+from datetime import datetime
+from typing import List
 
-class Metadata(DataSourceOutput):
-    """Class to store metedata data as a xr.Dataset with some validation"""
+import pandas as pd
+from pydantic import BaseModel, Field
 
-    __slots__ = ()
-    _expected_dimensions = ("t0_dt",)
+from nowcasting_dataset.consts import SPATIAL_AND_TEMPORAL_LOCATIONS_OF_EACH_EXAMPLE_FILENAME
+from nowcasting_dataset.filesystem.utils import check_path_exists
+from nowcasting_dataset.utils import get_start_and_end_example_index
 
-    # TODO: Add validation here - https://github.com/openclimatefix/nowcasting_dataset/issues/233
+
+class Metadata(BaseModel):
+    """Class to store metedata data"""
+
+    batch_size: int = Field(
+        ...,
+        g=0,
+        description="The size of this batch. If the batch size is 0, "
+        "then this item stores one data item",
+    )
+
+    t0_datetime_utc: List[datetime] = Field(
+        ...,
+        description="The t0s of each example ",
+    )
+
+    x_center_osgb: List[int] = Field(
+        ...,
+        description="The x centers of each example in OSGB coordinates",
+    )
+
+    y_center_osgb: List[int] = Field(
+        ...,
+        description="The y centers of each example in OSGB coordinates",
+    )
+
+    def save_to_csv(self, path):
+        """
+        Save metadata to a csv file
+
+        Args:
+            path: the path where the file shold be save
+
+        """
+
+        filename = f"{path}/{SPATIAL_AND_TEMPORAL_LOCATIONS_OF_EACH_EXAMPLE_FILENAME}"
+        metadata_dict = self.dict()
+        metadata_dict.pop("batch_size")
+
+        # if file exists, add to it
+        try:
+            check_path_exists(filename)
+        except FileNotFoundError:
+            metadata_df = pd.DataFrame(metadata_dict)
+
+        else:
+
+            metadata_df = pd.read_csv(filename)
+
+            metadata_df_extra = pd.DataFrame(metadata_dict)
+            metadata_df = metadata_df.append(metadata_df_extra)
+
+        metadata_df.to_csv(filename, index=False)
+
+
+def load_from_csv(path, batch_idx, batch_size) -> Metadata:
+    """
+    Load metadata from csv
+
+    Args:
+        path: the path which stores the metadata file
+        batch_idx: the batch index
+        batch_size: how many examples in each batch
+
+    Returns: Metadata class
+    """
+    filename = f"{path}/{SPATIAL_AND_TEMPORAL_LOCATIONS_OF_EACH_EXAMPLE_FILENAME}"
+
+    # read whole file
+    metadata_df = pd.read_csv(filename)
```

Comment:
if it's slow to read the entire CSV then consider using something like `read_csv(skiprows=start_example_idx+1, nrows=batch_size)`  (although I'm not sure if `skiprows` also skips the CSV header!)  But don't bother with this if reading the whole CSV file is fast enough!

Reply:
That seems worth doing